reusable_components_nlp/README.md

```diff
@@ -0,0 +1,989 @@
+namespace="kubeflow"
+
+sc = SeldonClient(
+    gateway="ambassador", 
+    ambassador_endpoint=host + ":" + port,
+    namespace=namespace)
+
+client_prediction = sc.predict(
+    data=batch, 
+    deployment_name=deployment_name,
+    names=["text"],
+    payload_type=payload_type,
+    transport="rest")
+
+print(client_prediction)
+```
+
+    Success:True message:
+    Request:
+    data {
+      names: "text"
+      ndarray {
+        values {
+          string_value: "Hello world this is a test"
+        }
+      }
+    }
+    
+    Response:
+    meta {
+      puid: "qtdca40d3s0463nn4ginhkvc6t"
+      routing {
+        key: "cleantext"
+        value: -1
+      }
+      routing {
+        key: "spacytokenizer"
+        value: -1
+      }
+      routing {
+        key: "tfidfvectorizer"
+        value: -1
+      }
+      requestPath {
+        key: "cleantext"
+        value: "clean_text_transformer:0.1"
+      }
+      requestPath {
+        key: "lrclassifier"
+        value: "lr_text_classifier:0.1"
+      }
+      requestPath {
+        key: "spacytokenizer"
+        value: "spacy_tokenizer:0.1"
+      }
+      requestPath {
+        key: "tfidfvectorizer"
+        value: "tfidf_vectorizer:0.1"
+      }
+    }
+    data {
+      names: "t:0"
+      names: "t:1"
+      ndarray {
+        values {
+          list_value {
+            values {
+              number_value: 0.6729318752883149
+            }
+            values {
+              number_value: 0.3270681247116851
+            }
+          }
+        }
+      }
+    }
+    
+
+
+## 6) Visualise Seldon's Production ML Pipelines
+We can visualise the performance using the SeldonAnalytics package, which we can deploy using:
+
+
+```python
+!helm install seldon-core-analytics --repo https://storage.googleapis.com/seldon-charts --namespace kubeflow
+```
+
+In my case, similar to what I did with Ambassador, I need to make sure the the service is a LoadBalancer instead of a NodePort
+
+
+```python
+!kubectl patch svc grafana-prom --type='json' -p '[{"op":"replace","path":"/spec/type","value":"LoadBalancer"}]' -n kubeflow
+```
+
+    service/grafana-prom patched
+
+
+
+```python
+!kubectl get svc grafana-prom -n kubeflow
+```
+
+    NAME           TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)        AGE
+    grafana-prom   LoadBalancer   10.98.248.223   localhost     80:32445/TCP   64m
+
+
+Now we can access it at the port provided, in my case it is http://localhost:32445/d/3swM2iGWz/prediction-analytics?refresh=5s&orgId=1
+
+(initial username is admin and password is password, which will be requested to be changed on the first login)
+
+Generate a bunch of requests and visualise:
+
+
+```python
+while True:
+    client_prediction = sc.predict(
+        data=batch, 
+        deployment_name=deployment_name,
+        names=["text"],
+        payload_type=payload_type,
+        transport="rest")
+```
+
+## You now have a full end-to-end training and production NLP pipeline 😎 
```

Comment:
Perhaps briefly explain what the graphs show here

Reply:
👍